src/qcommon/cm_load.c

```diff
@@ -456,15 +456,51 @@ void CMod_LoadBrushSides(lump_t *l)
 	}
 }
 
+/**
+ * @brief CMod_LoadCustomEntityString
+ * @param[in] name
+ */
+qboolean CMod_LoadCustomEntityString(const char* name) {
+	fileHandle_t file;
+	int entFileLen = 0;
+	char* filename;
+	char noext[MAX_QPATH];
+
+	COM_StripExtension(name, noext);
+	filename = va("%s.ent", noext);
+
+	entFileLen = FS_FOpenFileRead(filename, &file, qtrue);
+
+	if (file && entFileLen > 0)
+	{
+		cm.entityString = (char*)Hunk_Alloc(entFileLen + 1, h_high);
+		cm.numEntityChars = entFileLen + 1;
+		FS_Read(cm.entityString, entFileLen, file);
+		FS_FCloseFile(file);
+		cm.entityString[entFileLen] = '\0';
+		Com_Printf(va("Loaded entities from %s\n", filename));
```

Comment:
`va` should not be used for `Com_Printf`, it can handle it's formatting on it's own. Using `va` causes

```warning: format string is not a string literal (potentially insecure) [-Wformat-security]```